### PR TITLE
Fixed enchanted books with custom enchantments not working

### LINKED ISSUE #62: Custom enchantment books not working
Currently books with custom enchantments cannot be applied to the correct items in the anvil.

This would also be a good time to add in support for blacksmithing tables to be able to transfer over the custom enchantments

### PR DESCRIPTION
**Pull request by NRProjects**
Closes #62 

It does what it says on the tin. The smithing table event handler seems fairly simple, and I'm skeptical about its rigidity, but I can't think of a way it can be exploited or broken.

### CODE REVIEW

**Comment by paulpugh000** [comment]
I'm unable to do this review until #71 gets pulled in.

**Comment by Teelair** [comment]
> I'm unable to do this review until #71 gets pulled in.

You can change code without committing it to a branch. Hope that helps!

File: src/main/java/plugins/nate/smp/managers/EnchantmentManager.java [hunk 1]
```diff
@@ -194,47 +169,26 @@ public static void onAnvilPrepare(PrepareAnvilEvent event) {
...
-     */
     private static Map<Enchantment, Integer> getCustomEnchants(@NotNull ItemStack item) {
-        Map<Enchantment, Integer> enchants = new HashMap<>(item.getEnchantments());
-        List<Enchantment> toRemove = new ArrayList<>();
-        for (Enchantment enchant : enchants.keySet()) {
-            if (!ENCHANTMENTS.containsValue(enchant)) {
-                toRemove.add(enchant);
-            }
+        if (item.getItemMeta() instanceof EnchantmentStorageMeta meta) {
+
```
Comment (Teelair) [inline src/main/java/plugins/nate/smp/managers/EnchantmentManager.java:181]:
```suggestion
```

**Review by Teelair — APPROVED** [review]
Looks good to me. I have not tested, but will do before merging.

File: src/main/java/plugins/nate/smp/managers/EnchantmentManager.java [hunk 2]
```diff
@@ -274,3 +221,4 @@ private static List<String> getCustomLore(Map<Enchantment, Integer> enchantments
 }
 
 
+
```
Comment (Teelair) [inline src/main/java/plugins/nate/smp/managers/EnchantmentManager.java:224]:
```suggestion
```

**Pull request merged**